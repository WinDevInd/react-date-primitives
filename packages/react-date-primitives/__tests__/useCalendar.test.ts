import { renderHook, act } from 'react-hooks-testing-library';
import { useCalendar } from '../useCalendar';

import { isSameMonth } from '../utils';

describe('useCalendar hook tests', () => {
    test('gives correct days & month', () => {
        const date = new Date(2019, 0, 1, 0, 0, 0, 0);
        const { result } = renderHook(() => useCalendar(date));
        const days = result.current.days.map((week) =>
            week.map((day) => ({ ...day, date: day.date.toDateString() }))
        );

        expect(days).toMatchSnapshot();
        expect(isSameMonth(result.current.month, date)).toBe(true);
    });

    test('setMonth works', () => {
        const date1 = new Date(2019, 0, 1, 0, 0, 0, 0);
        const date2 = new Date(2019, 3, 1, 0, 0, 0, 0);
        const { result } = renderHook(() => useCalendar(date1));

        expect(isSameMonth(result.current.month, date1)).toBe(true);

        act(() => {
            result.current.setMonth(date2);
        });
        expect(isSameMonth(result.current.month, date2)).toBe(true);
    });
});
