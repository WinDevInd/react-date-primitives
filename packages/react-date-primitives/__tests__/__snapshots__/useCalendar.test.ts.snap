// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useCalendar hook tests gives correct days & month 1`] = `
Array [
  Array [
    Object {
      "date": "Sun Dec 30 2018",
      "inCurrentMonth": false,
    },
    Object {
      "date": "Mon Dec 31 2018",
      "inCurrentMonth": false,
    },
    Object {
      "date": "Tue Jan 01 2019",
      "inCurrentMonth": true,
    },
    Object {
      "date": "Wed Jan 02 2019",
      "inCurrentMonth": true,
    },
    Object {
      "date": "Thu Jan 03 2019",
      "inCurrentMonth": true,
    },
    Object {
      "date": "Fri Jan 04 2019",
      "inCurrentMonth": true,
    },
    Object {
      "date": "Sat Jan 05 2019",
      "inCurrentMonth": true,
    },
  ],
  Array [
    Object {
      "date": "Sun Jan 06 2019",
      "inCurrentMonth": true,
    },
    Object {
      "date": "Mon Jan 07 2019",
      "inCurrentMonth": true,
    },
    Object {
      "date": "Tue Jan 08 2019",
      "inCurrentMonth": true,
    },
    Object {
      "date": "Wed Jan 09 2019",
      "inCurrentMonth": true,
    },
    Object {
      "date": "Thu Jan 10 2019",
      "inCurrentMonth": true,
    },
    Object {
      "date": "Fri Jan 11 2019",
      "inCurrentMonth": true,
    },
    Object {
      "date": "Sat Jan 12 2019",
      "inCurrentMonth": true,
    },
  ],
  Array [
    Object {
      "date": "Sun Jan 13 2019",
      "inCurrentMonth": true,
    },
    Object {
      "date": "Mon Jan 14 2019",
      "inCurrentMonth": true,
    },
    Object {
      "date": "Tue Jan 15 2019",
      "inCurrentMonth": true,
    },
    Object {
      "date": "Wed Jan 16 2019",
      "inCurrentMonth": true,
    },
    Object {
      "date": "Thu Jan 17 2019",
      "inCurrentMonth": true,
    },
    Object {
      "date": "Fri Jan 18 2019",
      "inCurrentMonth": true,
    },
    Object {
      "date": "Sat Jan 19 2019",
      "inCurrentMonth": true,
    },
  ],
  Array [
    Object {
      "date": "Sun Jan 20 2019",
      "inCurrentMonth": true,
    },
    Object {
      "date": "Mon Jan 21 2019",
      "inCurrentMonth": true,
    },
    Object {
      "date": "Tue Jan 22 2019",
      "inCurrentMonth": true,
    },
    Object {
      "date": "Wed Jan 23 2019",
      "inCurrentMonth": true,
    },
    Object {
      "date": "Thu Jan 24 2019",
      "inCurrentMonth": true,
    },
    Object {
      "date": "Fri Jan 25 2019",
      "inCurrentMonth": true,
    },
    Object {
      "date": "Sat Jan 26 2019",
      "inCurrentMonth": true,
    },
  ],
  Array [
    Object {
      "date": "Sun Jan 27 2019",
      "inCurrentMonth": true,
    },
    Object {
      "date": "Mon Jan 28 2019",
      "inCurrentMonth": true,
    },
    Object {
      "date": "Tue Jan 29 2019",
      "inCurrentMonth": true,
    },
    Object {
      "date": "Wed Jan 30 2019",
      "inCurrentMonth": true,
    },
    Object {
      "date": "Thu Jan 31 2019",
      "inCurrentMonth": true,
    },
    Object {
      "date": "Fri Feb 01 2019",
      "inCurrentMonth": false,
    },
    Object {
      "date": "Sat Feb 02 2019",
      "inCurrentMonth": false,
    },
  ],
  Array [
    Object {
      "date": "Sun Feb 03 2019",
      "inCurrentMonth": false,
    },
    Object {
      "date": "Mon Feb 04 2019",
      "inCurrentMonth": false,
    },
    Object {
      "date": "Tue Feb 05 2019",
      "inCurrentMonth": false,
    },
    Object {
      "date": "Wed Feb 06 2019",
      "inCurrentMonth": false,
    },
    Object {
      "date": "Thu Feb 07 2019",
      "inCurrentMonth": false,
    },
    Object {
      "date": "Fri Feb 08 2019",
      "inCurrentMonth": false,
    },
    Object {
      "date": "Sat Feb 09 2019",
      "inCurrentMonth": false,
    },
  ],
]
`;
